pkg/app/services.go

```diff
@@ -224,3 +228,22 @@ func announceSystemServices[M ~map[string]T, T any](c *Controller, services *ser
 		c.Node.Announce(filepath.Join(c.Node.Name(), "systems"), node.HasClient(client)),
 	)
 }
+
+type serviceConfigStore struct {
+	store    serviceapi.Store
+	services *service.Map
+	id       string
+}
+
+func (s *serviceConfigStore) UpdateConfig(ctx context.Context, data []byte) error {
+	serv := s.services.Get(s.id)
+	if serv == nil {
+		return fmt.Errorf("service %s not found", s.id)
+	}
+	_, err := serv.Service.Configure(data)
+	if err != nil {
+		return err
+	}
```

Comment:
I'm not 100% sold on this bit. 

I think as the driver/etc will be the one calling this method, it already knows its config is updating and should handle it internally. I also think we might get issues here with ctx being cancelled before Save is called, it entirely depends on whether a driver/etc links the ctx given during `applyConfig` to the one passed to UpdateConfig.

If I'm thinking about it I think there are two separate concerns going on here:
1. The system wants to tell the driver the config has updated
2. The driver wants to tell the system the config has updated

Number 1 is for the general case: boot, generic APIs, getting config from the cloud, etc.
Number 2 is for the runtime case: submitting forms, discovery (of devices), user editing, etc.

Reply:
I see what you mean. It's more flexible if we let the service handle reconfiguring itself. 

I'll try to make this clear in the documentation.

Reply:
I've change the semantics of `ConfigUpdater` to not call `Configure()` any more.